### PR TITLE
delete item front-end

### PR DESCRIPTION
**Pull request by brandonktran**
swish

I had to move the total into app's state because whenever I try to delete an item right after I add one, the first item that I delete doesn't update the total, so the total is always one item off. The only fix I knew was to put it in state so that it will be guaranteed to be updated.

### CODE REVIEW

**Review by thebearingedge — CHANGES_REQUESTED** [review]
Ok two things:
1. I don't think putting the cart total in state is a real solution. Once you get your components back to dynamically rendering the cart total but not correctly, lemme know.
1. Try not to deviate from the styles you're already using. Your delete button should still be styled with Bootstrap.

**Comment by brandonktran** [comment]
swish

File: client/components/app.jsx [hunk 1]
```diff
@@ -60,11 +59,10 @@ export default class App extends React.Component {
             cart: newArray
           });
       }))
-      .then(data => this.calculateTotal());
+      .then(data => this.calculateTotal(this.state.cart));
```
Comment (thebearingedge) [inline client/components/app.jsx]: you can probably delete this line now

File: client/components/app.jsx [hunk 2]
```diff
@@ -41,7 +40,7 @@ export default class App extends React.Component {
     fetch('/api/cart')
       .then(res => res.json())
       .then(data => this.setState({ cart: data }))
-      .then(data => this.calculateTotal());
+      .then(data => this.calculateTotal(this.state.cart));
```
Comment (thebearingedge) [inline client/components/app.jsx]: you can probably delete this line now

File: client/components/app.jsx [hunk 3]
```diff
@@ -124,7 +123,7 @@ export default class App extends React.Component {
         <>
           <Header cartItemCount={this.state.cart.length} setView={this.setView} />
           <Transition key={this.state.view.name}>
-            <CartSummary cart={this.state.cart} setView={this.setView} total={this.total} deleteFromCart={this.deleteFromCart} />
+            <CartSummary cart={this.state.cart} setView={this.setView} total={this.calculateTotal} deleteFromCart={this.deleteFromCart} />
```
Comment (thebearingedge) [inline client/components/app.jsx]: Better to pass the component what it really wants (the total). so you can call this function here.

File: client/components/app.jsx [hunk 4]
```diff
@@ -133,7 +132,7 @@ export default class App extends React.Component {
         <>
           <Header cartItemCount={this.state.cart.length} setView={this.setView} />
           <Transition key={this.state.view.name}>
-            <CheckoutForm setView={this.setView} placeOrder={this.placeOrder} total={this.total} />
+            <CheckoutForm setView={this.setView} placeOrder={this.placeOrder} total={this.calculateTotal} cart={this.state.cart} />
```
Comment (thebearingedge) [inline client/components/app.jsx]: same here, the order form wants the total `this.calculateTotal()`

**Review by thebearingedge — CHANGES_REQUESTED** [review]

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**